session_pool.go

```diff
@@ -373,68 +367,57 @@ func (pool *SessionPool) getNextAddr() HostAddress {
 
 // getSession returns an available session.
 // This method should move an available session to the active list and should be MT-safe.
-func (pool *SessionPool) getIdleSession() (*Session, error) {
+func (pool *SessionPool) getIdleSession() (*pureSession, error) {
 	pool.rwLock.Lock()
 	defer pool.rwLock.Unlock()
 	// Get a session from the idle queue if possible
 	if pool.idleSessions.Len() > 0 {
-		session := pool.idleSessions.Front().Value.(*Session)
-		pool.removeSessionFromList(&pool.idleSessions, session)
-		pool.addSessionToList(&pool.activeSessions, session)
+		session := pool.idleSessions.Front().Value.(*pureSession)
+		pool.idleSessions.Remove(pool.idleSessions.Front())
 		return session, nil
 	} else if pool.activeSessions.Len() < pool.conf.maxSize {
-		// Create a new session if the total number of sessions is less than the max size
-		session, err := pool.newSession()
-		if err != nil {
-			return nil, err
-		}
-		pool.addSessionToList(&pool.activeSessions, session)
-		return session, nil
+		return nil, nil
```

Comment:
Why remove this? If `getIdleSession()` only checks the left session in the idle list, it's better to rename it to `getSessionFromIdle`